performance/swiftinterop/src/main/kotlin/org/jetbrains/multigraph/Multigraph.kt

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package org.jetbrains.multigraph
+
+interface Cost {
+    operator fun plus(other: Cost): Cost
+    operator fun minus(other: Cost): Cost
+    override operator fun equals(other: Any?): Boolean
+    operator fun compareTo(other: Cost): Int
+}
+
+data class Edge<T>(val id: UInt, val from: T, val to: T, val cost: Cost) {
+    override operator fun equals(other: Any?): Boolean {
+        return if (other is Edge<*>) (from == other.from && to == other.to && cost == other.cost) else false
+    }
+}
+
+class EdgeAbsenceMultigraphException(message: String): Exception(message) {}
+class VertexAbsenceMultigraphException(message: String): Exception(message) {}
+
+class Multigraph<T>() {
+    private var edges = mutableMapOf<T, MutableList<Edge<T>>>()
+    private var idCounter = 0u
+
+    val allVertexes: Set<T>
+        get() {
+            val outerVertexes = edges.keys
+            val innerVertexes = edges.map { (key, values) ->
+                values.map { it.to }.filter { it in outerVertexes }
```

Comment:
What is this code supposed to do?
As far as I see, `innerVertexes` is a subset of `outerVertexes`, so `outerVertexes.union(innerVertexes) == outerVertexes`.